contracts/examples/esdt-transfer-with-fee/src/esdt_transfer_with_fee.rs

```diff
@@ -0,0 +1,109 @@
+#![no_std]
+
+mod fee;
+use fee::*;
+
+elrond_wasm::imports!();
+#[elrond_wasm::derive::contract]
+pub trait EsdtTransferWithFee {
+    #[init]
+    fn init(&self) {}
+
+    #[only_owner]
+    #[endpoint]
+    fn set_general_fee(&self, fee: Fee<Self::Api>) {
+        self.general_fee().set(fee);
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn set_fee_for_token(&self, fee: Fee<Self::Api>, token: TokenIdentifier) {
+        self.specific_fee(&token).set(fee);
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn claim_fees(&self) {
+        let mut fees = ManagedVec::new();
+        for ((token, nonce), amount) in self.paid_fees().iter() {
+            fees.push(EsdtTokenPayment::new(token, nonce, amount))
+        }
+
+        let caller = self.blockchain().get_caller();
+        self.send().direct_multi(&caller, &fees, &[]);
+    }
+
+    #[payable("*")]
+    #[endpoint]
+    fn transfer(&self, address: ManagedAddress) {
+        require!(!self.general_fee().is_empty(), "Fees are not set yet");
+        let payments = self.call_value().all_esdt_transfers();
```

Comment:
What happens when someone is sending EGLD? Should the fees apply or should it be rejected?

Reply:
tokens that have no fee will be simply transfered withour expecting a fee

Reply:
This will not work for EGLD, as it's not transferable through the multi-transfer API.

Reply:
EGLD condition still missing. Try something like
`require!(self.call_value().egld_value() == 0, "EGLD transfers not allowed);`

Better framework syntax coming soon, but until then ....